### PR TITLE
Add update method, add storing instance data in meta

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

File: broker/controller.go [hunk 1]
```diff
@@ -197,6 +201,44 @@ func (c *Controller) CreateServiceInstance(w http.ResponseWriter, r *http.Reques
...
 }
 
+func (c *Controller) UpdateServiceInstance(w http.ResponseWriter, r *http.Request) {
+	log.Println("Update Service Instance...")
+
+	var instance ServiceInstance
+
+	err := ProvisionDataFromRequest(r, &instance)
+	if err != nil {
+		log.Println("Provision instatnce:", err)
```
Comment (dAdAbird) [inline broker/controller.go:226]: does it make sense to continue with the error?

File: broker/controller.go [hunk 2]
```diff
@@ -197,6 +201,44 @@ func (c *Controller) CreateServiceInstance(w http.ResponseWriter, r *http.Reques
...
+	}
+
+	instanceID := ExtractVarsFromRequest(r, "service_instance_guid")
+
+	skipS3 := true
+
+	err = c.DeployCluster(instance, &skipS3, instanceID)
```
Comment (dAdAbird) [inline broker/controller.go]: are you sure `DeployCluster `?
now both `UpdateServiceInstance` and `CreateServiceInstance` looks identical. is it ok?

File: broker/controller.go [hunk 3]
```diff
@@ -214,6 +256,46 @@ func (c *Controller) GetServiceInstance(w http.ResponseWriter, r *http.Request)
 	WriteResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) getBrokerInstances(typ string) {
```
Comment (dAdAbird) [inline broker/controller.go]: it's good to have unit tests for such type of functions

File: broker/controller.go [hunk 4]
```diff
@@ -214,6 +256,46 @@ func (c *Controller) GetServiceInstance(w http.ResponseWriter, r *http.Request)
...
+	s, err := dbaas.GetServiceBrokerInstances(typ)
+	if err != nil {
+		log.Println("getBrokerInstances:", err)
+		return
```
Comment (dAdAbird) [inline broker/controller.go]: return errors
don't log it everywhere

File: dbaas/psmdb/config.go [hunk 5]
```diff
@@ -622,6 +622,10 @@ func (cr *PerconaServerMongoDB) SetNew(clusterName, rsName string, c dbaas.Clust
 		return errors.Wrap(err, "new replset")
 	}
 
+	if len(c.PSMDB.BrokerInstance) > 0 {
+		cr.ObjectMeta.Annotations["broker-instance"] = c.PXC.BrokerInstance
```
Comment (dAdAbird) [inline dbaas/psmdb/config.go]: 1. make(map[string]string)?
2. **PXC** ? This is why I don't like current design where pxc and psmdb are put together, passed around and everybody knows about everything. That's a huge mess and leads to such kind of errors.

**Review by dAdAbird — CHANGES_REQUESTED** [review]

File: dbaas/psmdb/psmdb.go [hunk 6]
```diff
@@ -36,25 +36,36 @@ const (
 	defaultOperatorVersion = "percona/percona-server-mongodb-operator:1.1.0"
 )
 
+type Message struct {
+	Error Error `json:"error,omitempty"`
+}
+
+type Error struct {
+	Message string `json:"message"`
+}
```
Comment (dAdAbird) [inline dbaas/psmdb/psmdb.go]: I don't see how and where those two types are used

**Review by dAdAbird — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**